grama/eval_pnd.py

```diff
@@ -101,17 +125,18 @@ def eval_pnd(model, df_train, df_test, signs, n=int(1e4), seed=None, append=Fals
     ## Setup for reshaping
     means = []
     sds = []
-    vars = df_train.columns.values
-    input = model.var
-    outputs = [key for key in signs.keys() if key in df_test.columns.values]
-    length = int(len(model.out) / 2)
+    columns = df_train.columns.values
+    length = int(len(signs.keys()))
+    outputs = [key for key in signs.keys() if key in columns]
```

Comment:
Add an explicit check that the outputs implied by the `signs` argument are provided by the model; there's a lot that can go wrong, including the test you've implemented (`y` is not an output of the given model) and cases where the user mis-specifies a suffix (e.g. `y_mu` is the mean output, not `y_mean`).

Reply:
In this invariant check, make sure to give some more detailed information about what went wrong; for instance, something like

`signs argument "y" implies output "y_mean", which was not found in provided md.out`